src/tabs.tsx

```diff
@@ -117,26 +110,63 @@ export type TabsProps = {
 };
 
 const Tabs: React.FC<TabsProps> = ({selectedIndex, onChange, tabs}: TabsProps) => {
+    const id = useAriaId();
     const {width, ref} = useElementDimensions();
-    const classes = useStyles({width, numTabs: tabs.length});
+    const animatedLineRef = React.useRef<HTMLDivElement>(null);
+    const scrollableContainerRef = React.useRef<HTMLDivElement>(null);
+    const [isAnimating, setIsAnimating] = React.useState(false);
+    const classes = useStyles({width, numTabs: tabs.length, isAnimating});
+
+    const animateLine = (fromIndex: number, toIndex: number) => {
```

Comment:
This function animates the line:
1. sets initial line position and hides the line from the tab (setting the isAnimating state)
2. sets final line position and gets animated
3. once the animation ends, animated line is hidden and the line from the tab is shown (set isAnimating false)